lib/vanagon/driver.rb

```diff
@@ -76,15 +76,16 @@ def run
       @engine.startup(@workdir)
 
       puts "Target is #{@engine.target}"
-
-      install_build_dependencies
-      @project.fetch_sources(@workdir)
-      @project.make_makefile(@workdir)
-      @project.make_bill_of_materials(@workdir)
-      @project.generate_packaging_artifacts(@workdir)
-      @engine.ship_workdir(@workdir)
-      @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make})")
-      @engine.retrieve_built_artifact
+      Vanagon::Utilities.retry_with_timeout(3, 3600) do
+        install_build_dependencies
+        @project.fetch_sources(@workdir)
+        @project.make_makefile(@workdir)
+        @project.make_bill_of_materials(@workdir)
+        @project.generate_packaging_artifacts(@workdir)
+        @engine.ship_workdir(@workdir)
+        @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make})")
+        @engine.retrieve_built_artifact
+      end
```

Comment:
I would ultimately prefer that things that are inclined to fail have a retry (like cloning the github repo or downloading the source archive), but that also would be a lot of work to track that whole thing down